api/src/main/java/javax/enterprise/inject/spi/container/Container.java

```diff
@@ -0,0 +1,341 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.spi.container;
+
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.builder.AnnotatedTypeConfigurator;
+import javax.enterprise.inject.spi.builder.BeanConfigurator;
+import java.lang.annotation.Annotation;
+import java.util.Map;
+
+/**
+ * CDI container builder for Java SE.
+ * It is obtained by calling the {@link ContainerProvider#getContainer()} static method
+ * <p>
+ * <p>
+ * Typical usage looks like this:
+ * </p>
+ * <p>
+ * <pre>
+ * CDI<Object> cdi = ContainerProvider.getContainer().initialize();
+ * cdi.select(Foo.class).get();
+ * cdi.event().select(Bar.class).fire(new Bar());
+ * cdi.shutdown();
+ * </pre>
+ * <p>
+ * <p>
+ * The {@link CDI} interface implements AutoCloseable:
+ * </p>
+ * <p>
+ * <pre>
+ * try (CDI<Object> cdi = ContainerProvider.getContainer().initialize()) {
+ *     cdi.select(Foo.class).get();
+ * }
+ * </pre>
+ * <p>
+ * <p>
+ * By default, the discovery is enabled so that all beans from all discovered bean archives are considered. However, it's possible to define a "synthetic" bean
+ * archive, or the set of bean classes and enablement respectively:
+ * </p>
+ * <p>
+ * <pre>
+ * CDI<Object> cdi = ContainerProvider.getContainer().beanClasses(Foo.class, Bar.class).alternatives(Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * Moreover, it's also possible to disable the discovery completely so that only the "synthetic" bean archive is considered:
+ * </p>
+ * <p>
+ * <pre>
+ * CDI<Object> cdi = ContainerProvider.getContainer().disableDiscovery().beanClasses(Foo.class, Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * Container container = ContainerProvider.getContainer()
+ *    .disableDiscovery()
+ *    .packages(Main.class, Utils.class)
+ *    .interceptors(TransactionalInterceptor.class)
+ *    .property("property", true);
+ * CDI<Object> cdi = container.initialize();
+ * </pre>
+ *
+ * @since 2.0
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ */
+public interface Container {
+
+    /**
+     * Define the set of bean classes for the synthetic bean archive.
+     *
+     * @param classes classes to add to the synthetic bean archive
+     * @return self
+     */
+    Container beanClasses(Class<?>... classes);
```

Comment:
1. Built-in beans are always present; you never package them in your app, right?
2. `ContainerProvider.getContainer().addPackage().beanClasses()` - it's undefined right now, AFAIK Weld SE is processing packages and bean classes separately though (so there would be both, the package contents and classes)
3. If discovery is enabled, your classes reside in a bean archive and you also add them to the synthetic bean archive, then multiple beans will be registered for the same type. So yes, it's expected. Right now, Weld only logs a warning for each bean class deployed in multiple bean archives.